active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+Code a view-centric (Winforms-like) developer would write to enable Validation.
+
+
+### 1.10.1	C# Sample
+#### XAML
+
+```xaml
+<TextBox x:Name="Email" 
+         Header="Email" 
+         Validation.IsRequired="True" 
+         TextChanged="TextBox_TextChanged"/>
+```
+
+#### C#
+```csharp
+private void TextBox_TextChanged(object sender, TextChangedEventArgs e)
+{
+    WinformsValidate(sender, e);
+}
+
+private void WinformsValidate(object sender, TextChangedEventArgs e)
+{
+    var foo = sender as TextBox;
+    Validation.ClearErrors(foo);
+    Validation.AddErrorHandler(foo, OnValidationError);
+
+    if (!new EmailAddressAttribute().IsValid(foo.Text))
+    {
+
+        var ErrorsCollection = Validation.GetErrors(foo);
+        ErrorsCollection.Add(new ValidationError("Email is not a valid email address"));
+    }
+}
+
+private void OnValidationError(object sender, ValidationErrorEventArgs e)
+{
+    if (e.Action == ValidationErrorEventAction.Added)
+        // Do something when an error message was added
+    else
+        // Do something when an error was removed.
+
+        SaveButton.IsEnabled = Validation.GetErrors(e.Source);
+}
+```
+
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+
+# Remarks
+<!-- Explanation and guidance that doesn't fit into the Examples section. -->
+
+<!-- APIs should only throw exceptions in exceptional conditions; basically,
+only when there's a bug in the caller, such as argument exception.  But if for some
+reason it's necessary for a caller to catch an exception from an API, call that
+out with an explanation either here or in the Examples -->
+
+# API Notes
+<!-- Option 1: Give a one or two line description of each API (type
+and member), or at least the ones that aren't obvious
+from their name.  These descriptions are what show up
+in IntelliSense. For properties, specify the default value of the property if it
+isn't the type's default (for example an int-typed property that doesn't default to zero.) -->
+
+<!-- Option 2: Put these descriptions in the below API Details section,
+with a "///" comment above the member or type. -->
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+## InputPropertyAttribute
+The `InputPropertyAttribute` helps us ensure that we only generate validation code for the property specified by this attribute. This will help reduce code bloat by not generating code for properties that will never perform validation (like width). Controls can specify multiple properties as their input property. 
```

Comment:
Requiring an InputPropertyAttribute sounds difficult. Is there a way an x:Bind can specify it if the control author didn't realize it was required?

Reply:
One potential way we could do this is add a `Validation` value to the `BindingMode` enum:
`Text="{x:Bind Person.Name, Mode=Validation}"`

Or, we loosen the restriction. It might be unlikely that we'll be generating dead code, but it's hard to say with any certainty that we won't.

Reply:
@stevenbrix @MikeHillberg 

I associate the mode only with the direction of the binding, so a mode `Validation` looks a bit strange to me. Can't we have a `ValidatesOnNotifyDataErrors` property like on Binding in WPF:

```{x:Bind Person.Name, Mode=TwoWay,ValidatesOnNotifyDataErrors=True}```

Unlike in WPF, it could have the default value false. But: We could have the InputPropertyAttribute in addition. If you create a binding on such a property that has the InputPropertyAttribute, you have to explicitly set ValidatesOnNotifyDataErrors to false if you don't want to generate the binding code. By default, ValidatesOnNotifyDataErrors is true on an InputProperty binding

On the other side, if you have a property without the InputPropertyAttribute - the case that @MikeHillberg  mentioned - you could set ValidatesOnNotifyDataErrors to true to generate the validation code for that binding.

Reply:
The prospect of adding `Validation` as an option for  binding mode concerns me as it potentially causes the mode to have two meanings and would require changes in (especially 3rd party) tooling to handle the change in meaning.